lib/ansible/modules/cloud/amazon/route53_facts.py

```diff
@@ -342,7 +342,11 @@ def record_sets_details(client, module):
     elif module.params.get('type'):
         params['StartRecordType'] = module.params.get('type')
 
-    results = client.list_resource_record_sets(**params)
+    results = dict()
+    try:
+        results = client.list_resource_record_sets(**params)
```

Comment:
This is the only client method that has correct exception handling now. If the `to_native(e)` error messages are are clear, feel free just to move this try/except around `results = invocations[module.params.get('query')](route53, module)` in main() and fix all of them.

You could also remove `import boto` and the check for HAS_BOTO since it doesn't look like boto is used in this module, just boto3.